susemanager-utils/susemanager-sls/modules/engines/mgr_events.py

```diff
@@ -194,60 +203,58 @@ def _get_queue(self, minion_id):
-        return fnmatch.fnmatch(tag, "salt/job/*/ret/*") and self._is_salt_mine_update(
-            data
-        )
-
-    def _is_salt_mine_update(self, data):
-        return data.get("fun") == "mine.update"
-
-    def _is_presence_ping(self, tag, data):
-        return (
-            fnmatch.fnmatch(tag, "salt/job/*/ret/*")
-            and self._is_test_ping(data)
-            and self._is_batch_mode(data)
-        )
-
-    def _is_test_ping(self, data):
-        return data.get("fun") == "test.ping"
-
-    def _is_batch_mode(self, data):
-        return data.get("metadata", {}).get("batch-mode")
+    def push_events_from_queue(self):
+        while self._queue and (
+            self.counter < self._commit_burst or self._commit_burst == 0
+        ):
+            tag, data = self._queue.pop(0)
+            try:
+                self._insert(tag, data)
+            # pylint: disable-next=broad-exception-caught
+            except Exception:
+                # Exception while inserting the data, put the data back to the queue
+                # and repeat the attempt of pushing it with the next cycle of queue_thread
+                # The exception was logged before in _insert
+                self._queue.insert(0, (tag, data))
+                break
 
     @salt.ext.tornado.gen.coroutine
     def add_event_to_queue(self, raw):
-        # FIXME: Drop once we only use Salt >= 3004
-        if salt.version.SaltStackVersion(*salt.version.__version_info__).major < 3004:
-            tag, data = self.event_bus.unpack(raw, self.event_bus.serial)
-        else:
+        try:
             tag, data = self.event_bus.unpack(raw)
-        self._insert(tag, data)
+        # pylint: disable-next=broad-exception-caught
+        except Exception as e:
+            log.warning("Unable to unpack the event data: %s", e)
+        self._queue.append((tag, data))
 
     def db_keepalive(self):
         if self.connection.closed:
-            log.error("%s: Diconnected from database. Trying to reconnect...", __name__)
+            log.error("Diconnected from database. Trying to reconnect...")
             self._connect_to_database()
 
-    @salt.ext.tornado.gen.coroutine
-    def add_token(self):
-        self.tokens = min(self.tokens + 1, self.config["commit_burst"])
-        self.attempt_commit()
-        self.trace_log()
-        self.event_bus.io_loop.call_later(
-            self.config["commit_interval"], self.add_token
-        )
+    def queue_thread(self):
+        while True:
+            try:
+                self.db_keepalive()
+                self.push_events_from_queue()
+                self.trace_log()
+                self.attempt_commit()
+            # pylint: disable-next=broad-exception-caught
+            except Exception as e:
+                log.error(
+                    "Exception while processing the events queue: %s", e, exc_info=True
+                )
+            time.sleep(self._commit_interval)
```

Comment:
this thread sleep always happens, even if we have event on the queue to get processed. I think we could have a different approach, and commit data after a `_commit_burst` or `_commit_interval` is reached. What do you think?